src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -5681,17 +5681,8 @@ private static BigDecimal divideAndRound128(final long dividendHi, final long di
         tmp = (dividendHi << shift) | (dividendLo >>> 64 - shift);
         long u2 = tmp & LONG_MASK;
         long q1, r_tmp;
```

Comment:
Nit: There doesn't seem to be a reason for splitting the q1 and r\_tmp declarations from their assignments, why not the following instead? (Also applies to the cases below)
```
long q1 = Long.divideUnsigned(tmp, v1);
long r_tmp = Long.remainderUnsigned(tmp, v1);
```